connectors/tests/test_utils.py

```diff
@@ -51,13 +53,65 @@ def test_invalid_names():
             validate_index_name(name)
 
 
+@pytest.mark.asyncio
+async def test_mem_queue_speed(patch_logger):
+    # with memqueue
+    queue = MemQueue(maxmemsize=1024 * 1024, refresh_interval=0.1, refresh_timeout=2)
+    start = time.time()
+    for i in range(1000):
+        await queue.put("x" * 100)
+    mem_queue_duration = time.time() - start
+
+    # vanilla queue
+    queue = asyncio.Queue()
+    start = time.time()
+    for i in range(1000):
+        await queue.put("x" * 100)
+    queue_duration = time.time() - start
+
+    # mem queue should be 30 times slower at the most
+    quotient, _ = divmod(mem_queue_duration, queue_duration)
+    assert quotient < 30
+
+
+@pytest.mark.asyncio
+async def test_mem_queue_race(patch_logger):
+    item = "small stuff"
+    queue = MemQueue(
+        maxmemsize=get_size(item) * 2 + 1, refresh_interval=0.1, refresh_timeout=30
+    )
+    max_size = 0
```

Comment:
I think a comment would help to understand what's going on in the test. I vaguely understand it, but not 100% sure and not 100% sure what is the actual problem.